### PR TITLE
Fixed the problem that the script callback function could not be registered in game.cpp

### PR DESCRIPTION
**Pull request by qiuguohua**
1. Add engine start event
2. When the engine is created, create a script engine at the same time

### CODE REVIEW

File: native/cocos/engine/Engine.cpp [hunk 1]
```diff
@@ -117,8 +119,9 @@ int32_t Engine::init() {
 #if CC_USE_PROFILER
     _profiler = ccnew Profiler();
 #endif
-
-    _scriptEngine = ccnew se::ScriptEngine();
+    if (!_scriptEngine) {
+        _scriptEngine = ccnew se::ScriptEngine();
+    }
```
Comment (minggo) [inline native/cocos/engine/Engine.cpp]: Please add comments why need to do it like this.

Comment (qiuguohua) [reply native/cocos/engine/Engine.cpp]: OK

Comment (minggo) [reply native/cocos/engine/Engine.cpp]: > The script engine is created in the engine constructor for the first time.

Please add a reason for it.

Comment (dumganhar) [reply native/cocos/engine/Engine.cpp]: Could users's custom JSB registeration in Game.cpp be invoked again while restarting?

**Review by dumganhar — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Pull request merged**